test/test_spectral_ops.py

```diff
@@ -510,6 +510,119 @@ def test_fftn_invalid(self, device):
         with self.assertRaisesRegex(RuntimeError, "Expected a real input"):
             torch.fft.rfftn(c)
 
+    # 2d-fft tests
+
+    # NOTE: 2d transforms are only thin wrappers over n-dim transforms,
+    # so don't require exhaustive testing.
+
+    @skipCPUIfNoMkl
+    @skipCUDAIfRocm
+    @onlyOnCPUAndCUDA
+    @unittest.skipIf(not TEST_NUMPY, 'NumPy not found')
+    @dtypes(torch.double, torch.complex128)
+    def test_fft2_numpy(self, device, dtype):
+        norm_modes = ((None, "forward", "backward", "ortho")
+                      if LooseVersion(np.__version__) >= '1.20.0'
+                      else (None, "ortho"))
+
+        # input_ndim, s
+        transform_desc = [
+            *product(range(2, 5), (None, (4, 10))),
+        ]
+
+        fft_functions = ['fft2', 'ifft2', 'irfft2']
+        if dtype.is_floating_point:
+            fft_functions += ['rfft2']
+
+        for input_ndim, s in transform_desc:
+            shape = itertools.islice(itertools.cycle(range(4, 9)), input_ndim)
+            input = torch.randn(*shape, device=device, dtype=dtype)
+            for fname, norm in product(fft_functions, norm_modes):
+                torch_fn = getattr(torch.fft, fname)
+                numpy_fn = getattr(np.fft, fname)
+
+                def fn(t: torch.Tensor, s: Optional[List[int]], dim: List[int] = (-2, -1), norm: Optional[str] = None):
+                    return torch_fn(t, s, dim, norm)
+
+                torch_fns = (torch_fn, torch.jit.script(fn))
+
+                # Once with dim defaulted
+                input_np = input.cpu().numpy()
+                expected = numpy_fn(input_np, s, axes=(-2, -1), norm=norm)
```

Comment:
This is interesting. Should this not explicitly provide a value for axes to be consistent with the actual computation below?